packages/@jsii/go-runtime-test/project/callbacks_test.go

```diff
@@ -7,20 +7,21 @@ import (
 )
 
 func TestPureInterfacesCanBeUsedTransparently(t *testing.T) {
-	expected := calc.StructB{RequiredString: "It's Britney b**ch!"}
-	delegate := &StructReturningDelegate{expected: expected}
-	consumer := calc.NewConsumePureInterface(delegate)
+	requiredstring := "It's Britney b**ch!"
+	expected := calc.StructB{RequiredString: &requiredstring}
+	delegate := StructReturningDelegate{expected: &expected}
+	consumer := calc.NewConsumePureInterface(&delegate)
```

Comment:
There appears to be a bug related to registering managed objects when this is passed by value instead of by reference. The only relevant change here is to the logic within the CastPtrToRef.

This also feels related to #2686 where structs passed by value that are fields of structs, and/or items within a slice, aren't being initialized correctly. Began hunting this down but need some more time.